pkg/plugins/gateway/algorithms/pd_disaggregation.go

```diff
@@ -174,17 +184,23 @@ func (r *pdRouter) filterPrefillDecodePods(routingCtx *types.RoutingContext, rea
 			continue
 		}
 
+		if aibrixPromptLengthBucketing {
+			if !r.isPodSuitableForPromptLength(pod, promptLength) {
```

Comment:
Assuming, if no pod was suitable for prompt length then final prefillPods array will be empty.

Better alternative will be

```
var prefillPodsByPromptLength []*v1.Pod

for _, pod := range readyPods {
	role := pod.Labels[PDRoleIdentifier]

	switch role {
	case "prefill":
		prefillPods = append(prefillPods, pod)

		if aibrixPromptLengthBucketing && r.isPodSuitableForPromptLength(pod) {
			prefillPodsByPromptLength = append(prefillPodsByPromptLength, pod)
		}

	case "decode":
		decodePods = append(decodePods, pod)
	}
}

// Override prefill pods only if bucketing produced results
if aibrixPromptLengthBucketing && len(prefillPodsByPromptLength) > 0 {
	prefillPods = prefillPodsByPromptLength
}
```